File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = arr.size();
        vector<int>prefixXor(n+1);
        for(int i=0; i<n; i++) {
            prefixXor[i+1] = prefixXor[i] ^ arr[i];
        }
        vector<int>ans;
        for(auto q:queries) {
            ans.push_back(prefixXor[q[1]+1] ^ prefixXor[q[0]]);
        }
        return ans;
    }
};

File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int[] temp = new int[n+1];
        int[] ans = new int[queries.length];
        temp[0] = 0;
        for(int i=1; i<=n; i++) {
            temp[i] = temp[i-1] ^ arr[i-1];
        }
        for(int i=0; i<queries.length; i++) {
            ans[i] = temp[queries[i][0]] ^ temp[queries[i][1] + 1];
        }
        return ans;
    }
}